online_meetings/bluejeans_report.py

```diff
@@ -33,141 +36,175 @@
-        for csv_file in glob.glob("./meetings_*.csv"):
-            logger.info(csv_file)
-            df = pd.read_csv(csv_file)
-
-            # append to the total output
-            total_df = total_df.append(df)
-            logger.info(df)
-
-            # remove file
-            os.remove(csv_file)
-
-    # remove zip file
-    os.remove(zip_file_name)
-    print(f" {zip_file_name} file Removed!")
-    
-
-# Remove any duplicate uuids in the record
-logger.info(f"Initial dataframe size: {len(total_df)}")
-if "meetingUUID" in total_df:
-    total_df.drop_duplicates("meetingUUID", inplace=True)
-    logger.info(f"Dataframe with duplicates removed: {len(total_df)}")
-total_df.to_csv("./bluejeans_meetings.csv")
+        
+        reportJobs_response = requests.request("POST", url, headers=headers, data=payload)
+        reportJob_json = json.loads(reportJobs_response.text.encode('utf8'))
+        if ("status" in reportJob_json) and (reportJob_json["status"] == "success") and ("jobId" in reportJob_json):
+            # get the job id
+            job_id = reportJob_json["jobId"]
+            logger.info("report job id = " + job_id)
+
+        # sleep to wait for report job finish
+        download_file_path = ""
+        while not download_file_path :
+            # retrieve report job information including the file download path
+            jobs_url = f"https://indigo-api.bluejeans.com/v1/enterprise/{ENV.get('BLUEJEANS_ENTERPRISE_ID')}/indigo/jobs/list?userid={ENV.get('BLUEJEANS_USER_ID')}"
+            payload = {}
+            headers = {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json',
+                'Authorization': f'Bearer {token}'
+            }
+            jobs_response = requests.request("GET", jobs_url, headers=headers, data=payload)
+            jobs_json = json.loads(jobs_response.text.encode('utf8'))
+            for job_json in jobs_json:
+                # find the right job:
+                if job_json['_id'] == job_id:
+                    if (job_json['status'] == 'success'):
+                        download_file_path = job_json['message']['path']
+                        logger.info(f"job {job_id} finished yet: {job_json['status']} and download file path is {download_file_path}")
+                        break
+                    else:
+                        logger.info(f"job {job_id} not finished yet: {job_json['status']}")
+                        time.sleep(10)
+
+        download_url = f"https://indigo-api.bluejeans.com/v1/enterprise/{ENV.get('BLUEJEANS_ENTERPRISE_ID')}/indigo/{download_file_path}"
+
+        # download the report file
+        payload = {}
+        headers = {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json',
+            'Authorization': f'Bearer {token}'
+        }
+        r = requests.get(download_url, headers=headers)
+
+        # specifying the zip file name
+        zip_file_name = "./bj.zip"
```

Comment:
will this file name is always going be like `bj.zip`? what is the purpose of adding the `./ ` to this? If I have to run this locally do I need to have this located physically on my machine?

Reply:
If the file name is going to be same all the time then make it a constant